validator/sawtooth_validator/execution/scheduler_parallel.py

```diff
@@ -36,43 +36,44 @@
 
 
 class AddressNotInTree(Exception):
-    pass
+    def __init__(self, match=None):
+        super().__init__()
+        self.match = match
 
 
 class Node:
     def __init__(self, address, data=None):
         self.address = address
-        self.children = {}
+        self.children = set()
         self.data = data
 
     def __repr__(self):
-        return repr({
-            'address': self.address,
-            'children': self.children,
-            'data': self.data,
-        })
+        return repr((
+            self.address,
+            self.data,
+            self.children,
+        ))
 
 
 class Tree:
-    def __init__(self, token_size):
+    def __init__(self):
```

Comment:
Great job writing a very descriptive commit message. It would probably be a good idea to include some of that description about how addresses are represented in this Tree in a docstring under the `class Tree:` line for future reference.